src/hotspot/share/gc/shenandoah/shenandoahGeneration.hpp

```diff
@@ -135,10 +135,6 @@ class ShenandoahGeneration : public CHeapObj<mtGC>, public ShenandoahSpaceInfo {
   void increase_capacity(size_t increment);
   void decrease_capacity(size_t decrement);
 
-  void set_soft_max_capacity(size_t soft_max_capacity) {
-    _soft_max_capacity = soft_max_capacity;
```

Comment:
Not only the accessor, but it seems like the `_soft_max_capacity` field in `ShenandoahGeneration` is vestigial and can be deleted?

A cursory look seemed to indicate that a field with a similar name (similar accessor, but confusingly different field name) is used for the purpose for that purpose for the whole heap, and the generation's field is obsolete/vestigial. I may have missed some uses in my inspection, but you can check if deleting the field is ok (it's only used in logging where it seems not to have much value?).

Reply:
Hmm, this field is used by the heuristics. This looks like a regression to me. The soft max heap size can be given as a command line argument, but it is also supposed to be a 'managed' option that can be changed with jmx. I will open a separate ticket for this.